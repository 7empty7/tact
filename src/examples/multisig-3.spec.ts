import { MultisigContract_init } from "./output/multisig-3_MultisigContract";
import { createExecutorFromCode } from 'ton-nodejs';
import { Address, Cell, contractAddress } from "ton";
import BN from "bn.js";

describe('muiltisig-3', () => {
    it('should deploy', async () => {

        let key1 = new BN(1);
        let key2 = new BN(2);
        let key3 = new BN(3);
        let res = await MultisigContract_init(key1, key2, key3);
        expect(res.data.toDebugString()).toMatchSnapshot();

        let executor = await createExecutorFromCode(res);
        expect((await executor.get('seqno')).stack.readBigNumber().toString(10)).toBe('0');

        expect((await executor.get('key1')).stack.readBigNumber().toString(10)).toBe('1');
        expect((await executor.get('key2')).stack.readBigNumber().toString(10)).toBe('2');
        expect((await executor.get('key3')).stack.readBigNumber().toString(10)).toBe('3');

        // let tp = (await executor.get('testAddress')).stack.readTuple();
        // let n = tp.readNumber();
        // let n2 = tp.readBigNumber();
        // let a = new Address(n, Buffer.from(n2.toString(16).padStart(64, '0'), 'hex'));
        // console.warn(a.toFriendly());
        // let addr = contractAddress({ workchain: 0, initialCode: new Cell(), initialData: new Cell() });
        // console.warn(addr.toFriendly());
    });
});