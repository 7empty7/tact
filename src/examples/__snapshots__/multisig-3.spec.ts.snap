// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`muiltisig-3 should deploy 1`] = `
"x{00000000000000000000000000000000000000000000000000000000000000000000000100000000000000000000000000000000000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000003}
 x{C_}
  x{D04_}
   x{FF00F4A413F4BCF2C80B}
    x{62_}
     x{CB}
      x{2_}
       x{D4_}
        x{1C0875D27087E54C0835C2C7F780B434C0C05C6C3000645FE45C38807E900C1510455BC0FE1840A456F820840C378A50AEB8C08C3CB0192_}
         x{ED44D0D401F862D31FD3FFD3FFD3FF55306C1404D31F01821030DE2942BAF2E064D31FFA00FA4001433003D401D001D401D001D401D0164330361089107810675504F013C8F84201CC55305034CB1FCBFFCBFFCBFFC9ED54}
        x{081BBC93A_}
       x{5}
        x{2_}
         x{321C4072804572801C0072809400F3C5807E809C00729A1C007280089BACE65FC0728000BC005633254C9C16328038B2407EC02_}
         x{1C013233154C140D32C7F2FFF2FFF2FFF26_}
        x{2_}
         x{1FCCC05C1B7C032_}
         x{0408D7C0E_}
      x{62_}
       x{2_}
        x{04D7C0E_}
        x{1B0C6_}
       x{2_}
        x{17C0E_}
        x{151D50F215481408F2C7C07E808073C5B27E4014810ABE44148CCA7E441504CDFE4414D66EBCA280AC006C3CA2BC03A_}
     x{2_}
      x{BE64BF6A2686A00FC31698FE9FFE9FFE9FFAA98360A78094_}
      x{2_}
       x{2_}
        x{2_}
         x{B0E9FB513435007E18B4C7F4FFF4FFF4FFD54C1B053C046_}
         x{B0E1BB513435007E18B4C7F4FFF4FFF4FFD54C1B053C042_}
        x{B47CBDA89A1A803F0C5A63FA7FFA7FFA7FEAA60D829E01F_}
       x{B8AC7F00D}
"
`;
